web/src/stream_edit.ts

```diff
@@ -618,7 +619,35 @@ export function initialize(): void {
             data.description = new_description;
         }
 
-        dialog_widget.submit_api_request(channel.patch, url, data);
+        dialog_widget.submit_api_request(channel.patch, url, data, {
+            success_continuation() {
+                dialog_widget.close();
+            },
+            error_continuation(xhr) {
+                const $modal = $("#change_stream_info_modal");
+
+                const {code} = z.object({code: z.string()}).parse(xhr.responseJSON);
+
+                if (code === "CHANNEL_ALREADY_EXISTS") {
+                    $modal.find("#dialog_error").hide().empty();
+
+                    assert(data.new_name !== undefined);
+                    const existing_stream = stream_data.get_sub_by_name(data.new_name);
+
+                    if (existing_stream) {
+                        const rendered_error = render_channel_name_conflict_error({
+                            stream_id: existing_stream.stream_id,
+                            is_archived: existing_stream.is_archived,
+                            show_rename: false,
+                            can_view_channel: true,
+                        });
+
+                        $modal.find("#stream_name_error").html(rendered_error).show();
```

Comment:
Here also `$modal.find` might be needed if you use a unique ID for this error element.